src/solve/infer/mod.rs

```diff
@@ -45,31 +39,21 @@ impl InferenceTable {
         self.lifetime_unify.new_key(InferenceValue::Unbound(ui))
     }
 
-    pub fn new_krate_variable(&mut self, ui: UniverseIndex) -> KrateInferenceVariable {
-        self.krate_unify.new_key(InferenceValue::Unbound(ui))
-    }
-
     pub fn new_parameter_variable(&mut self, ui: ParameterKind<UniverseIndex>)
                                   -> ParameterInferenceVariable {
         match ui {
             ParameterKind::Ty(ui) => ParameterKind::Ty(self.new_variable(ui)),
             ParameterKind::Lifetime(ui) => ParameterKind::Lifetime(self.new_lifetime_variable(ui)),
-            ParameterKind::Krate(ui) => ParameterKind::Krate(self.new_krate_variable(ui)),
         }
     }
 
     pub fn snapshot(&mut self) -> InferenceSnapshot {
         let ty_unify_snapshot = self.ty_unify.snapshot();
-        let krate_unify_snapshot = self.krate_unify.snapshot();
-        InferenceSnapshot {
-            ty_unify_snapshot,
-            krate_unify_snapshot,
-        }
+        InferenceSnapshot { ty_unify_snapshot }
```

Comment:
Hmm, this seems like a pre-existing bug, but shouldn't there be a `lifetime_snapshot` in this struct too...? I guess I forgot to add one.